### PR TITLE
Improved design and added  caching with redis

### PR DESCRIPTION
**Pull request by BrainIsDead**
Relosve: [SD-17](https://cmu-delphi.atlassian.net/browse/SD-17), [SD-18](https://cmu-delphi.atlassian.net/browse/SD-18)

- Added linkpreview, pymemcache, and django-redis to Pipfile
- Modified models.py to add a get_preview method to the Link model that returns a dictionary containing information about the link preview
- Modified migrations/0002_alter_link_link_type.py to remove unique=True from the link_type field of the Link model
- Added migrations/0003_alter_geography_options_alter_signaltype_options_and_more.py to modify the ordering and verbose_name_plural options for Geography and SignalType models, and remove unique=True from base field of Signal model.
- Modified src/templates/signals/signal_detail.html to update the layout and add badges for signal properties, as well as a section for related links.
- Modified src/templates/signals/signal_list.html to make table rows clickable and add tooltips for selecting multiple sources in the filter dropdown.
- Added redis service to docker-compose.yaml
- Modified settings.py to add Django cache configuration for Redis
- Modified views.py to add caching for SignalsDetailView using Django's cache_page decorator

### CODE REVIEW

**Comment by krivard** [comment]
What is caching needed for?

File: Dockerfile [hunk 1]
```diff
@@ -3,6 +3,9 @@ FROM python:3.10.8-alpine
 ENV PYTHONDONTWRITEBYTECODE=1
 ENV PYTHONUNBUFFERED=1
 RUN addgroup -S python && adduser -S python -G python
+RUN apk update
+RUN apk add musl-dev mariadb-dev gcc
+RUN apk add graphviz-dev
```
Comment (krivard) [inline Dockerfile:8]: What is graphviz being used for? I see it in the Pipfile but it doesn't seem to be mentioned anywhere else.

**Review by krivard — APPROVED** [review]
👍 

I don't love using `<br>` to add vertical whitespace but I don't feel strongly about it

File: Dockerfile (hunk 1 above)
Comment (BrainIsDead) [reply Dockerfile:8]: It's part of django extensions used to visualize DB models like I shared before, I don't know why but pygraphviz starts to fail installig without installed pacckage graphviz-dev

**Pull request merged**